### PR TITLE
Feature/cicd cloudbuild

### PR DESCRIPTION
**Pull request by danistrebel**
What's changed, or what was fixed?

- CI/CD reference for both Jenkins (SaaS) and Cloud Build (SaaS/hybrid/X)
- Note: Leaving the pipeline for `cicd-jenkins` because the PR builder fails upon removed dirs

- [x] I have run all the tests locally and they all pass.
- [x] I have followed the relevant style guide for my changes.

**CC:** @apigee-devrel-reviewers

### CODE REVIEW

**Comment by danistrebel** [comment]
Please note that I had to add a "[N/A]" pass for pipelines if the pipeline.sh does not exist. Our current process with `git diff` would not allow for removed or renamed projects otherwise. This will have an impact on strictly enforcing the existence of pipeline.sh files. On the other hand it makes it more visible in the CI output than a silent no-op pipeline.sh.

**Comment by seymen** [comment]
Would it make sense the structure the folders a little bit differently now that we have two separate ci configurations in the same project?

My proposal:
```
- airports-v1
  - apiproxy
  - test
  - pom.xml
  - package.json
  - package-lock.json
  - eslint
- jenkins-build
  - img
  - .dockerignore
- ci-config
  - jenkins
    - Jenkinsfile
  - cloudbuild
    - cloudbuild.yaml
    - cloudignore
```

We can also create links directly to the `jenkins` or `cloudbuild` configuration files from the README.

**Comment by danistrebel** [comment]
Folders re-organized.

/gcbrun

File: references/cicd-pipeline/ci-config/cloudbuild/cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+#  limitations under the License.
+
+steps:
+  # Install node dependencies for linting and testing
+- name: node
+  id: "Install Dependencies"
+  entrypoint: npm
+  args:
+    - "install"
+    - "--slient"
```
Comment (laughingbiscuit) [inline references/cicd-pipeline/ci-config/cloudbuild/cloudbuild.yaml:22]: silent

File: references/cicd-pipeline/ci-config/jenkins/Jenkinsfile [hunk 2]
```diff
@@ -27,7 +27,7 @@ pipeline {
                   // env.APIGEE_ORG = 'apigee-org-name'
               // All other branches are deployed as separate proxies with suffix in the test environment
               } else {
-                  env.APIGEE_DEPLOYMENT_SUFFIX = env.GIT_BRANCH ? "-" + env.GIT_BRANCH.replaceAll("\\W", "-") : "-jenkins"
+                  env.APIGEE_DEPLOYMENT_SUFFIX = env.GIT_BRANCH ? "-" + env.GIT_BRANCH.replaceAll("\\W", "-") : "--devrel-jenkins"
```
Comment (laughingbiscuit) [inline references/cicd-pipeline/ci-config/jenkins/Jenkinsfile:30]: trivial: one dash

**Review by laughingbiscuit — APPROVED** [review]

File: references/cicd-pipeline/Jenkinsfile [hunk 3]
```diff
@@ -96,38 +90,31 @@ pipeline {
...
-        }
 
-        stage('Deploy proxy bundle') {
-          steps {
-            sh "mvn apigee-enterprise:deploy -B -P${env.MAVEN_PROFILE} -Dapigee.org=${env.APIGEE_ORG} -Dapigee.username=${APIGEE_CREDS_USR} -Dapigee.password=${APIGEE_CREDS_PSW} -Ddeployment.suffix=${env.APIGEE_DEPLOYMENT_SUFFIX}"
+            sh """
+              mvn clean install \
+                -P"apigeeapi,${env.MAVEN_PROFILE}" \
+                -Dpassword="${APIGEE_CREDS_PSW}" \
+                -Dusername="${APIGEE_CREDS_USR}" \
```
[the remark this answers is not in the document]
Comment (danistrebel) [reply references/cicd-pipeline/Jenkinsfile]: They unfortunately come from Jenkins credentials https://github.com/jenkinsci/pipeline-examples/blob/master/declarative-examples/simple-examples/credentialsUsernamePassword.groovy

**Pull request merged**